pkg/util/dns.go

```diff
@@ -0,0 +1,57 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/miekg/dns"
+)
+
+const defaultResolvConf = "/etc/resolv.conf"
+
+// GetDefaultDNSServers returns the DNS servers from the default
+// resolver configuration.
+func GetDefaultDNSServers() ([]string, error) {
+	conf, err := dns.ClientConfigFromFile(defaultResolvConf)
+	if err != nil {
+		return nil, err
+	}
+	return conf.Servers, nil
+}
+
+// LookupSRV tries to resolve an SRV query of the given service, proto and domain name.
+// proto can be 'tcp' or udp'.
+// The query will be of the form _service._proto.name.
+func LookupSRV(service, proto, name string) ([]*dns.SRV, error) {
+	dnsServers, err := GetDefaultDNSServers()
+	if err != nil {
+		return nil, err
+	}
+
+	msg := &dns.Msg{}
+	name = "_" + service + "._" + proto + "." + name
+	msg.SetQuestion(dns.Fqdn(name), dns.TypeSRV)
+
+	var result []*dns.SRV
+	dnsResolved := false
+	for _, serverAddr := range dnsServers {
+		resMsg, err := dns.Exchange(msg, serverAddr)
+		if err != nil {
+			continue
```

Comment:
Please log the error here. Deploying this, gives this error: `dial udp: address 10.47.240.10: missing port in address`